### PR TITLE
Fix #1305 InsertAfter method throws if dub attribute is inserted at the end of the collection.

### LINKED ISSUE #1305: Inconsistent behavior of XmlAttributeCollection.InsertBefore and InsertAfter
I've noticed an inconsistent behavior of `XmlAttributeCollection.InsertAfter(XmlAttribute newNode, XmlAttribute refNode)` comparing to `InsertBefore` when the reference node is the last in the collection (the first for `InsertBefore`) and it's a duplicate of the new node. The duplicate node (attribute) needs to be removed before inserting. `InsertAfter` is unable to handle the case and throws `System.ArgumentOutOfRangeException` further in `List<T>.Insert()` due to an incorrectly calculated insert position, while `InsertBefore` does the job. I'm not sure if this case has any practical use, but might be worth fixing.

The following unit test reproduces the problem:

        [Fact]
        // [ActiveIssue("Throws argument out of range exception")]
        public void InsertAfterRemovesDupRefAttr()
        {
            const string attributeName = "existingAttr";
            const string attributeUri = "some:existingUri";
            XmlDocument doc = CreateDocumentWithElement();
            XmlElement element = doc.DocumentElement;
            element.Attributes.Append(doc.CreateAttribute("attr1", "some:uri1"));
            element.Attributes.Append(doc.CreateAttribute("attr2", "some:uri2"));
            XmlAttribute refAttr = element.Attributes.Append(doc.CreateAttribute(attributeName, attributeUri)); //dup
            XmlAttribute newAttr = doc.CreateAttribute(attributeName, attributeUri);

            XmlAttributeCollection target = element.Attributes;
            target.InsertAfter(newAttr, refAttr); // throws

            // Assert...
        }

A similar case (the reference attribute is the first in the collection) works for `InsertBefore`:

        [Fact]
        public void InsertBeforeRemovesDupRefAttr()
        {
            const string attributeName = "existingAttr";
            const string attributeUri = "some:existingUri";
            XmlDocument doc = CreateDocumentWithElement();
            XmlElement element = doc.DocumentElement;
            XmlAttribute refAttr = element.Attributes.Append(doc.CreateAttribute(attributeName, attributeUri)); //dup
            element.Attributes.Append(doc.CreateAttribute("attr1", "some:uri1"));
            element.Attributes.Append(doc.CreateAttribute("attr2", "some:uri2"));
            XmlAttribute newAttr = doc.CreateAttribute(attributeName, attributeUri);

            XmlAttributeCollection target = element.Attributes;
            target.InsertBefore(newAttr, refAttr);

            // Assert...
        }

### PR DESCRIPTION
**Pull request by heXelium**
@krwq Can you please check?

### CODE REVIEW

File: src/System.Xml.XmlDocument/tests/XmlAttributeCollectionTests/InsertAfterTests.cs [hunk 1]
```diff
@@ -208,5 +208,26 @@ public void InsertAfterReturnsInsertedAttr()
...
+        public void InsertAfterRemovesDupRefAttrAtTheEnd()
+        {
+            const string attributeName = "existingAttr";
+            const string attributeUri = "some:existingUri";
+            XmlDocument doc = CreateDocumentWithElement();
+            XmlElement element = doc.DocumentElement;
+            XmlAttribute anotherAttr1 = element.Attributes.Append(doc.CreateAttribute("attr1", "some:uri1"));
+            XmlAttribute anotherAttr2 = element.Attributes.Append(doc.CreateAttribute("attr2", "some:uri2"));
+            XmlAttribute refAttr = element.Attributes.Append(doc.CreateAttribute(attributeName, attributeUri)); //dup
+            XmlAttribute newAttr = doc.CreateAttribute(attributeName, attributeUri);
```
Comment (krwq) [inline src/System.Xml.XmlDocument/tests/XmlAttributeCollectionTests/InsertAfterTests.cs]: Could you also add a test like:
nodeA
nodeB
nodeRef
nodeC

and InsertAfter duplicate of NodeRef after NodeRef? Also please test if inserting dup of nodeRef after nodeC and nodeB if those are not already present

**Comment by krwq** [comment]
Thank you for this fix! Overall looks good, although please add few more tests.

**Comment by heXelium** [comment]
Thank you. Added two more tests. Cases of inserting dup of nodeRef after nodeC and nodeB I believe is already covered by **InsertAfterRemovesDupAttrBeforeTheRef** and **InsertAfterRemovesDupAttrAfterTheRef**.

**Comment by stephentoub** [comment]
LGTM.  Thanks, @heXelium.

**Pull request merged**